### PR TITLE
feat: add TeX wrapping functionality to processResponseContent

### PR DESCRIPTION
**Pull request by Locutus7**
https://dsai.atlassian.net/browse/CHAT-1724

### Problem

Cohere Command models (and potentially others) occasionally emit LaTeX math commands directly in their markdown output **without surrounding math delimiters**. For example:

Final Answer:
\boxed{4\ \text{gallons}}

The KaTeX marked extension only activates on explicit delimiters (`$...$`, `$$...$$`, `\(...\)`, `\[...\]`). Without them, `\boxed{4\ \text{gallons}}` falls through as plain text and renders literally in the UI instead of as a formatted math expression.

### Root Cause

This is a joint model/client issue. The model outputs valid LaTeX syntax but omits the delimiters that signal to the renderer that math is intended. The client has no mechanism to detect and recover from this.

### Solution

Added `wrapUndelimitedTexExpressions()` in `src/lib/utils/index.ts`, called from `processResponseContent()` before the marked lexer runs.

The function:
1. **Splits the content** into protected segments (code fences, inline code, already-delimited math) and plain text, so existing rendering is never disturbed
2. **Applies a single-pass regex** across unprotected segments to detect bare math-exclusive TeX commands (`\boxed`, `\frac`, `\sqrt`, `\binom`, `\dfrac`, `\tfrac`, `\underbrace`, `\overbrace`) with up to two levels of brace nesting
3. **Wraps matched expressions** in `$...$` so KaTeX picks them up normally

A single-pass combined regex is used (rather than sequential replacements) to prevent double-wrapping when commands are nested - e.g. `\boxed{\frac{1}{2}}` must become `$\boxed{\frac{1}{2}}$`, not `$\boxed{$\frac{1}{2}$}$`.

### Tests

Unit tests added in `src/lib/utils/index.test.ts` (run with `npm run test:frontend`):

- Wraps `\boxed{...}`, `\frac{...}{...}`, `\sqrt{...}` when bare
- Does **not** double-wrap already-delimited math (`$...$`, `$$...$$`)
- Does **not** touch TeX inside code blocks or inline code
- Handles nested commands (`\boxed{\frac{1}{2}}`) without double-wrapping

### Files Changed

- `src/lib/utils/index.ts` — `processResponseContent` now pre-processes content through `wrapUndelimitedTexExpressions`
- `src/lib/utils/index.test.ts` — new Vitest unit tests

### CODE REVIEW

File: src/lib/utils/index.ts [hunk 1]
```diff
@@ -203,8 +203,48 @@ export const sanitizeResponseContent = (content: string): string => {
...
+	const B = `\\{(?:[^{}]|\\{(?:[^{}]|\\{[^{}]*\\})*\\})*\\}`;
+
+	// Single-pass combined regex — prevents double-wrapping when commands are nested
+	// (e.g. \boxed{\frac{1}{2}} would be double-wrapped by two sequential replacements).
+	// Alternation order matters: the longer/outer command is tried first so the inner
+	// occurrence is consumed as part of it rather than matched separately.
+	const COMBINED_RE = new RegExp(
+		`\\\\(?:` +
+			// Single brace group: \boxed, \sqrt, \binom family, \underbrace, \overbrace, \dfrac, \tfrac
+			`(?:boxed|dfrac|tfrac|binom|dbinom|tbinom|underbrace|overbrace|sqrt)\\s*${B}` +
```
Comment (pascal-beaulne) [inline src/lib/utils/index.ts:231]: @handshape should we test for all possibilities or simply displaying that one or 2 works it's good enough for you?

Comment (handshape) [reply src/lib/utils/index.ts:231]: I'd make sure the test covers the depth of the coverage we're tying to hit -- 
at least one doubly-nested expression. 
One expression in a code fence.
One already-delimited expression.

Comment (handshape) [reply src/lib/utils/index.ts:231]: Looks like a lot of that is already covered.

Comment (handshape) [reply src/lib/utils/index.ts:231]: Mmm. It's the bare stuff that's uncovered.

Energy-mass equivalence: E = mc^2
Summation: \sum_{i=1}^{n} x_i
Definite integral: \int_{a}^{b} g(t) \, dt
Greek letters & relations: \alpha + \beta = \gamma

This gets real messy the more we look at it.

Comment (handshape) [reply src/lib/utils/index.ts:231]: Okay. Just did a survey of the non-Cohere cases that got emitted today, and they're all wrapped in a top level math container of some kind. Let's forget the bare stuff for the purposes of this unit of work. The edge cases around the regex are worth trying out.

We're using client-side TeX rendering, so the damage an adversary can do with nasty content is limited.

We can test just one or two of the top-level container expressions, @pascal-beaulne .

Comment (Locutus7) [reply src/lib/utils/index.ts:231]: Let me know if you want me to add more tests in order to move this PR along. It covers most things as you saw while solving the core issue raised in the ticket.

**Review by pascal-beaulne — APPROVED** [review]

File: src/lib/utils/index.ts (hunk 1 above)
Comment (handshape) [reply src/lib/utils/index.ts:231]: Per the talk yesterday during the backlog refinement, this is resolved.
